src/emission_profile.jl

```diff
@@ -0,0 +1,82 @@
+#imports
+using Plots
+
+#functions
+"""
+From Reynolds (2020) (Eq2-4)
+Calculates the radius of the inner most stable circular orbit of the a black hole,
+with spin a_star, and mass M.
+"""
+function r_isco(a_star, M)
+    
+    a = a_star
+    r_g = 2*G*M/(c^2)
+    z1 = 1+∛(1-a^2)*(∛(1+a)+∛(1-a))
+    z2 = √(3*a^2+z1^2)
+    if a >= 0
+        r_isco = (3+z2-(3-z1)*√(3+z1+2*z2))*r_g
+    elseif a < 0
+        r_isco = (3+z2+(3-z1)*√(3+z1+2*z2))*r_g
+    end
+end
+
+"""
+From Page & Thorne (1974) (Eq15n)
+Calculates the function f at radius r, for a black hole of spin a_star, and mass M, 
+which is used to calculate the flux given by its accretion disk.
+"""
+function f(r, a_star, M)
+
+    R_isco = r_isco(a_star, M)
+  
+    x = √(r/M)
+    x0 = √(R_isco/M)
+    x1 = 2*cos((1/3)*(acos(a_star))-(π/3))
+    x2 = 2*cos((1/3)*(acos(a_star))+(π/3))
+    x3 = -2*cos((1/3)*(acos(a_star)))
+
+    f = (3/(2*M))*(1/(x^2*(x^3-(3*x)+(2*a_star))))*(x-x0-((3/2)*a_star*log(x/x0)) 
+            - (3*(x1-a_star)^2)/(x1*(x1-x2)*(x1-x3))*log((x-x1)/(x0-x1)) 
+            - (3*(x2-a_star)^2)/(x2*(x2-x1)*(x2-x3))*log((x-x2)/(x0-x2)) 
+            - (3*(x3-a_star)^2)/(x3*(x3-x1)*(x3-x2))*log((x-x3)/(x0-x3)))
+end
```

Comment:
This function keeps hitting a domain error
```julia
DomainError with -9.736039653587063e-28:
    sqrt will only return a complex result if called with a complex argument. Try sqrt(Complex(x)).
    Stacktrace:
      [1] throw_complex_domainerror(f::Symbol, x::Float64)
        @ Base.Math ./math.jl:33
      [2] sqrt(x::Float64)
        @ Base.Math ./math.jl:567
      [3] f(r::Float64, a_star::Float64, M::Float64)
        @ AccretionFormulae ~/Developer/DevEnv/dev/AccretionFormulae/src/emission_profile.jl:33
      [4] diss(mdot::Float64, r::Float64, a_star::Float64, M::Float64)
  ```
From the readout, it's specifically line 33, and from the magnitude of the value, it looks like there's some scaling factor missing.

Reply:
We just tested the r_isco function, and got these results
![M =M_sun](https://user-images.githubusercontent.com/94124028/154975975-e6c42902-6ecd-4915-85b9-625c0a6519e5.png)
M = M_sun

![M = 1](https://user-images.githubusercontent.com/94124028/154975997-b6a04439-8ddb-41e7-a1ac-df5d73239cf9.png)
M = 1

For M = M_sun, it seems that the negative value comes when the magnitude of a = 1, which seems to be the default setting in `test.jl`. We can add something to r_isco which could either return an error or truncate the value when the magnitude of a > 0.998 (max spin of a physical black hole). 

For the second one, we set M = 1 as used in `test.jl`, and it seems to give the scaling error you mentioned. For your `CarterMethodBL()` function, are we correct in thinking that the spin, a, is measured in units of M, and M in solar masses? I think this might give us some problems as in our functions, it uses SI units to give us correctly scaled outputs.

Reply:
We are just going through all of our equations now to try and figure out if all our units are consistent, so we might find some mistakes if they are there.